### ISSUE TITLE
REMOVE_ENCHANT x KEEP_ON_DEATH bug

### ISSUE THREAD

**Issue by SirJben** [GC-spigot/AdvancedEnchantments#3849]
### Describe the bug

REMOVE_ENCHANT does not work if you are not wearing the item. It will work only in armor slots but the rest with items that has soulbound in inventory it will not work and the enchantment stays there.

### How to reproduce

```soulboundtest:
  display: '%group-color%Soulbound'
  description: You may not lost your item on death
  applies-to: any armor or weapon
  type: DEATH;PASSIVE_DEATH
  group: SIMPLE
  applies:
  - PLAYER_HEAD
  - ALL_TOOLS
  - ALL_SWORD
  - ALL_AXE
  - ALL_ARMOR
  - BOW
  - CROSSBOW
  levels:
    '1':
      chance: 100
      effects:
      - REMOVE_ENCHANT:soulboundtest
      - KEEP_ON_DEATH```

### Screenshots / Videos

before death*
![image](https://github.com/GC-spigot/AdvancedEnchantments/assets/63577950/a9739790-c641-4358-85de-cf26acbfeed8)
after death**
![image](https://github.com/GC-spigot/AdvancedEnchantments/assets/63577950/c397dd27-f7e5-4821-a901-cfdf26ee2019)
![image](https://github.com/GC-spigot/AdvancedEnchantments/assets/63577950/3134b93a-4bc0-41a5-ad12-1d88110f8105)




### "/ae plinfo" link

https://paste.md-5.net/etizamepiq

### Server Log

_No response_

**Comment by xlsalvolx** [comment]
Can confirm the issue

**Comment by xlsalvolx** [comment]
Also, there is also this big issue with the keep_on_death effect that it won't proc with special kind of damages. For example, in mythicmobs you can deal damage that ignores armors (will bypass armor defence) and once killed the keep_on_death effect won't recognize it as death and won't activate

How to reproduce:
1. Install any version of mythicmobs
2. Add this mob as a .txt in the "mobs" folder of mythicmobs: https://paste.md-5.net/tecafadede.coffeescript
3. Get killed by the mob
4. Soulbound won't activate

**Comment by xlsalvolx** [comment]
Why this was marked as fixed? NOTHING got fixed on 9.5.23

https://www.youtube.com/watch?v=GZ5T0XrqNlA&feature=youtu.be

The enchant is still not getting removed and it gets removed after several deaths on most items

**Comment by xlsalvolx** [comment]
```display: '%group-color%Soulbound'
  description: You may not lost your item on death
  applies-to: any armor or weapon
  type: DEATH;PASSIVE_DEATH
  group: SIMPLE
  applies:
  - PLAYER_HEAD
  - ALL_TOOLS
  - ALL_SWORD
  - ALL_AXE
  - ALL_ARMOR
  - BOW
  - CROSSBOW
  levels:
    '1':
      chance: 100
      effects:
      - REMOVE_ENCHANT:soulboundtest
      - KEEP_ON_DEATH```

I'm still using the same config
